Source/AliveLib/MineCar.cpp

```diff
@@ -972,206 +982,184 @@ void MineCar::State_1()
     field_C4_velx = FP_FromInteger(0);
     field_C8_vely = FP_FromInteger(0);
 
-    // TODO: Refactor
-    bool enterRightBlock = false;
-    if (!sInputObject_5BD4E0.isPressed(sInputKey_Right_5550D0))
-    {
-        if (!sInputObject_5BD4E0.isPressed(field_1D4_previous_input))
-        {
-            enterRightBlock = true;
-        }
-
-        if ((unsigned __int16)field_1D6_continue_move_input != sInputKey_Right_5550D0)
+    InputCommands inputKey = sInputKey_Right_5550D0;
+
+    if ((sInputObject_5BD4E0.isPressed(inputKey) ||
+        (sInputObject_5BD4E0.isPressed(field_1D4_previous_input) &&
+        (unsigned __int16) field_1D6_continue_move_input == inputKey &&
+        field_1D4_previous_input != (unsigned __int16) inputKey &&
+        field_turnDirection_1BC != MineCarDirs::eUp_3 &&
+        field_turnDirection_1BC != MineCarDirs::eDown_0))
+        && !(WallHit_408750(mineCarHeight * FP_FromDouble(0.5), mineCarWidthAdjusted + FP_FromInteger(1)) &&
+        WallHit_408750(mineCarHeight - FP_FromInteger(8), mineCarWidthAdjusted + FP_FromInteger(1))))
+    {
+        FP rayCast1 = (kGridSize / FP_FromInteger(4)) + field_B8_xpos;
```

Comment:
nearly all these locals can be const, also would it not be clearer to just pass them directly maybe?